dotnet/src/Microsoft.AutoGen/Runtime/WorkerGatewayService.cs

```diff
@@ -8,6 +8,17 @@ internal sealed class WorkerGatewayService(WorkerGateway agentWorker) : AgentRpc
 {
     public override async Task OpenChannel(IAsyncStreamReader<Message> requestStream, IServerStreamWriter<Message> responseStream, ServerCallContext context)
     {
-        await agentWorker.ConnectToWorkerProcess(requestStream, responseStream, context);
+        try
+        {
+            await agentWorker.ConnectToWorkerProcess(requestStream, responseStream, context);
```

Comment:
What type of exception will it throw if `context.CancellationToken.IsCancellationRequested` is true

Reply:
no exception - close the stream. quesion is should we return a cancellation token so that if callers are awaiting this stream they know to cancel? I think there are bugs in here.